Velaptor/Content/Caching/SoundCache.cs

```diff
@@ -0,0 +1,184 @@
+        /// <summary>
+        /// Finalizes an instance of the <see cref="SoundCache"/> class.
+        /// </summary>
+        [ExcludeFromCodeCoverage]
+        ~SoundCache()
+        {
+            if (UnitTestDetector.IsRunningFromUnitTest)
+            {
+                return;
+            }
+
+            ShutDown();
+        }
+
+        /// <inheritdoc/>
+        public int TotalCachedItems => this.sounds.Count;
+
+        /// <inheritdoc/>
+        public ReadOnlyCollection<string> CacheKeys => new (this.sounds.Keys.ToArray());
+
+        /// <summary>
+        /// Gets a sound using the given <paramref name="soundFilePath"/>.
+        /// </summary>
+        /// <param name="soundFilePath">
+        ///     The full file path to a <c>Sound</c> file.
+        /// </param>
+        /// <returns>A sound loaded from either a <c>'.ogg'</c> or <c>'.mp3'</c> file.</returns>
+        /// <remarks>
+        /// <para>
+        ///     If the item has not been previously created, the <see cref="TextureCache"/> class
+        ///     will retrieve it, and then cache it for fast retrieval.
+        /// </para>
+        /// <para>
+        ///     The only sound files supported are <c>'.ogg'</c> and <c>'.mp3'</c> files.
+        /// </para>
+        /// </remarks>
+        /// <exception cref="ArgumentNullException">
+        ///     Thrown if the <paramref name="soundFilePath"/> is null or empty.
+        /// </exception>
+        /// <exception cref="LoadSoundException">
+        ///     Thrown if the type of sound file being requested is not any of the supported types below:
+        /// <list type="bullet">
+        ///     <item><c>'.ogg'</c></item>
+        ///     <item><c>'.mp3'</c></item>
+        /// </list>
+        /// </exception>
+        /// <exception cref="FileNotFoundException">
+        ///     Thrown if the file at the given <paramref name="soundFilePath"/> is not found.
+        /// </exception>
+        public ISound GetItem(string soundFilePath)
+        {
+            if (string.IsNullOrEmpty(soundFilePath))
+            {
+                throw new ArgumentNullException(nameof(soundFilePath), "The parameter must not be null or empty.");
+            }
+
+            var extension = this.path.GetExtension(soundFilePath);
+
+            if (new[] { OggFileExtension, Mp3FileExtension }.Contains(extension) is false)
+            {
+                var exceptionMsg = $"Sound file type '{extension}' is not supported.";
+                exceptionMsg += $"\nSupported file types are '{OggFileExtension}' and '{Mp3FileExtension}'.";
+
+                throw new LoadSoundException(exceptionMsg);
+            }
+
+            var cacheKey = soundFilePath;
+
+            if (this.file.Exists(soundFilePath))
+            {
+                return this.sounds.GetOrAdd(cacheKey, filePath => this.soundFactory.Create(filePath));
+            }
+
+            throw new FileNotFoundException($"The '{extension}' sound file does not exist.");
```

Comment:
Send in the file path as one of the parameters into this `FileNotFoundException` ctor

```suggestion
            throw new FileNotFoundException($"The '{extension}' sound file does not exist.", soundFilePath);
```